templates/sfx-core.js

```diff
@@ -367,7 +367,7 @@
     entry.module.execute.call(global);
   }
 
-  var nodeRequire = typeof System != 'undefined' && System._nodeRequire || typeof require != 'undefined' && require.resolve && typeof process != 'undefined' && require;
+  var nodeRequire = typeof System != 'undefined' && System._nodeRequire || typeof process != 'undefined' && !process.browser && typeof require != 'undefined' && typeof require.resolve != 'undefined' && require;
```

Comment:
Here are the things I changed:

1) We now only set nodeRequire to be the `require` variable when `!process.browser`. As far as I can tell, process.browser seems to be a browserify thing that webpack and other build systems have also adopted. So I'm hoping it is the right thing to check, since we do not want to set `nodeRequire` to be the nodejs `require` if our code is being bundled for the browser. I feel pretty good about that for the UMD case, but am not as sure for the other SFX use cases.
2) I moved the `typeof process` check to the beginning, so that the process.browser check can be near the beginning so that we short circuit out of the conditional expression before hitting the `&& require` that webpack replaces with a `throw` statement. This is what fixes the runtime error that is thrown by webpack (webpack replaces `&& require` with a `throw` statement.
3) I changed `&& require.resolve` to `&& typeof require.resolve != 'undefined'` because webpack knows that a typeof statement doesn't count as a dynamic usage of `require.resolve`. This is what eliminates one of the webpack build-time warnings.